dbms/include/DB/IO/ReadHelpers.h

```diff
@@ -619,6 +619,10 @@ inline void readBinary(UInt8 & x, 	ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(UInt16 & x, 	ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(UInt32 & x, 	ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(UInt64 & x, 	ReadBuffer & buf) { readPODBinary(x, buf); }
+#ifdef __APPLE__
```

Comment:
Should have comment with detailed explanation (in one place - here - will be enough):

"On Linux x86_64 int64_t maps to long, but on Apple x86_64, it maps to long long: two different types; but on both platforms, Int64 maps to long."